clients/src/test/java/org/apache/kafka/common/protocol/MessageUtilTest.java

```diff
@@ -83,4 +90,31 @@ public void testConstants() {
         assertEquals(MessageUtil.UNSIGNED_SHORT_MAX, 0xFFFF);
         assertEquals(MessageUtil.UNSIGNED_INT_MAX, 0xFFFFFFFFL);
     }
+
+    @Test
+    public void testBinaryNode() throws IOException {
+        byte[] expected = new byte[] {5, 2, 9, 4, 1, 8, 7, 0, 3, 6};
+        StringWriter writer = new StringWriter();
+        ObjectMapper mapper = new ObjectMapper();
+
+        mapper.writeTree(mapper.createGenerator(writer), new BinaryNode(expected));
+
+        JsonNode binaryNode = mapper.readTree(writer.toString());
+
+        assertTrue(binaryNode.isTextual(), binaryNode.toString());
```

Comment:
can you make the assertion failure message a little bit clearer? Just printing the node seems a bit cryptic.